Packages/com.unity.inputsystem/InputSystem/Devices/Keyboard.cs

```diff
@@ -779,4 +777,72 @@ public void OnIMECompositionChanged(IMECompositionString compositionString)
 
         internal InlinedArray<Action<IMECompositionString>> m_ImeCompositionListeners;
     }
+
+    public static class KeyboardExtensions
+    {
+        public static bool IsModifierKey(this Key key)
+        {
+            switch (key)
+            {
+                case Key.LeftAlt:
+                case Key.RightAlt:
+                case Key.LeftShift:
+                case Key.RightShift:
+                case Key.LeftMeta:
+                case Key.RightMeta:
+                case Key.LeftCtrl:
+                case Key.RightCtrl:
+                    return true;
+            }
+            return false;
+        }
+
+        ////REVIEW: Is this a good idea? Ultimately it's up to any one keyboard layout to define this however it wants.
+        public static bool IsTextInputKey(this Key key)
+        {
+            switch (key)
+            {
+                case Key.LeftAlt:
+                case Key.RightAlt:
+                case Key.LeftShift:
+                case Key.RightShift:
+                case Key.LeftMeta:
+                case Key.RightMeta:
+                case Key.LeftCtrl:
+                case Key.RightCtrl:
+                case Key.ContextMenu:
+                case Key.Escape:
+                case Key.Delete:
```

Comment:
Why is delete note a "text input key"? are the arrow keys "text input keys"?

Reply:
Not in the context of this function. I think the problem may be the name of the function. Maybe `GeneratesPrintableTextCharacter` is better?

What the function tries to establish is whether a key is considered to have a printable character representation or is considered a "function" key. Bit hard to establish unambiguously. Enter/return, for example, clearly generates text character input -- just not the kind being looked for here.